colcon_cmake/task/cmake/__init__.py

```diff
@@ -208,3 +209,53 @@ def get_visual_studio_version():
     :rtype: str
     """
     return os.environ.get('VisualStudioVersion', None)
+
+
+def _parse_cmake_version(version_str):
+    """
+    Parse the version number line from running 'cmake --version'.
+
+    The output is given as described in get_cmake_version().
+
+    :returns: The version number split as described above.
+    :rtype (int, int, int, str)
+    """
+    # Version number is on the first line. We can ignore anything else.
+    ver_re_str = \
+        r'^(?:cmake\s+version)?\s+([0-9]+)\.([0-9]+)\.([0-9]+)(?:\-(\S+))?'
+    if version_str and len(version_str):
+        ver_match = re.match(ver_re_str, version_str)
+        if ver_match:
+            return (int(ver_match.group(1)), int(ver_match.group(2)),
+                    int(ver_match.group(3)), ver_match.group(4))
+
+    # Failed to parse the version number.
+    return (0, 0, 0, 'failed')
```

Comment:
Instead of implementing a custom function how about using `pkg_resources.parse_version` instead?

Reply:
Good to know.

Reply:
Right. In which case the following test will fail because the major part of the version is less than 3. At any rate, this is moot with the change to using pkg_resources for version parsing.